src/components/layout/ScrollScreen.tsx

```diff
@@ -36,6 +36,7 @@ export interface ScrollScreenBaseProps {
   contentContainerStyle?: StyleProp<ViewStyle>
   fill?: boolean
   headerOptions?: HeaderOptions
+  headerProps?: Partial<BaseHeaderProps>
```

Comment:
Hm, this is a bit confusing. We are now passing both `headerOptions` which is part of `headerProps`. So if `options` is included in the `headerProps` it will override whatever is passed in `headerOptions`.

Reply:
Currently our `HeaderOptions` and `BaseHeaderProps` look like this:

```ts
export type HeaderOptions = Pick<StackHeaderProps['options'], 'headerRight' | 'headerLeft' | 'headerTitle'> & {
  type?: 'default' | 'stack'
}

export interface BaseHeaderProps extends ViewProps {
  options: HeaderOptions
  headerTitleRight?: () => ReactNode
  ...
```

How about moving `headerTitleRight` into `HeaderOptions`?

```ts
export type HeaderOptions = Pick<StackHeaderProps['options'], 'headerRight' | 'headerLeft' | 'headerTitle'> & {
  headerTitleRight?: () => ReactNode
  type?: 'default' | 'stack'
}

export interface BaseHeaderProps extends ViewProps {
  options: HeaderOptions
  ...
```

That way, we can simplify the `DashboardScreenStyled` component from

```ts
<DashboardScreenStyled
      refreshControl={
        <RefreshSpinner refreshing={isLoading} onRefresh={refreshData} progressViewOffset={headerHeight || 170} />
      }
      hasBottomBar
      verticalGap
      headerOptions={{
        headerRight: () => <DashboardHeaderActions />,
        headerLeft: () => <WalletSwitchButton isLoading={isLoading} />,
        headerTitle: walletName
      }}
      headerProps={{
        headerTitleRight: () => (
          <NetworkBadgeContainer>
            <ActiveNetworkBadge />
          </NetworkBadgeContainer>
        )
      }}
      {...props}
    >
```

to

```ts
    <DashboardScreenStyled
      refreshControl={
        <RefreshSpinner refreshing={isLoading} onRefresh={refreshData} progressViewOffset={headerHeight || 170} />
      }
      hasBottomBar
      verticalGap
      headerOptions={{
        headerRight: () => <DashboardHeaderActions />,
        headerLeft: () => <WalletSwitchButton isLoading={isLoading} />,
        headerTitle: walletName,
        headerTitleRight: () => (
          <NetworkBadgeContainer>
            <ActiveNetworkBadge />
          </NetworkBadgeContainer>
        )
      }}
      {...props}
    >
```